applications/tari_conduit/src/main.rs

```diff
@@ -0,0 +1,630 @@
+extern crate arrayref;
+extern crate chrono;
+extern crate jsonrpc;
+use chrono::Local;
+use curl::{
+    easy::{Auth, Easy, List},
+    Error,
+};
+
+use log::*;
+use monero::{
+    blockdata::{
+        block::BlockHeader,
+        transaction::{SubField, Transaction},
+        Block,
+    },
+    consensus::{deserialize, encode::VarInt, serialize},
+    cryptonote::hash::{Hash, Hashable},
+};
+use rand::rngs::OsRng;
+use regex::Regex;
+use serde::{Deserialize, Serialize};
+use std::{
+    convert::TryFrom,
+    io::{prelude::*, Read},
+    net::{TcpListener, TcpStream},
+};
+use tari_core::{
+    blocks::NewBlockTemplate,
+    consensus::{ConsensusManager, ConsensusManagerBuilder, Network},
+    mining::CoinbaseBuilder,
+    transactions::types::CryptoFactories,
+};
+use tari_crypto::{keys::SecretKey, ristretto::RistrettoSecretKey};
+use tari_utilities::ByteArray;
+use tokio::runtime::Runtime;
+
+mod grpc_conversions;
+
+pub type PrivateKey = RistrettoSecretKey;
+
+pub mod grpc {
+    tonic::include_proto!("tari.base_node"); // The string specified here must match the proto package name
+}
+
+// TODO: Refactor into a configuration
+const MONEROD_URL: &str = "monero-stagenet.exan.tech:38081";
+const MONEROD_USER: &str = "user";
+const MONEROD_PASS: &str = "pass";
+const TARI_GRPC_URL: &str = "http://127.0.0.1:18142";
+const USE_AUTH: bool = false;
+
+// TODO: Refactor this to be shared
+#[derive(Clone, Debug)]
+pub struct TransientData {
+    tari_block: Option<grpc::GetNewBlockResult>,
+    monero_seed: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, Default)]
+pub struct MoneroData {
```

Comment:
Is there a reason we are copying this struct, rather than import it from the tari::core?

Reply:
tari::core got the updated version of this from the conduit once I got it working. Will import form there..